src/main/java/org/elasticsearch/kafka/indexer/service/ElasticSearchBatchService.java

```diff
@@ -57,40 +59,41 @@ private void initBulkRequestBuilder(){
      */
     public void addEventToBulkRequest(String inputMessage, String indexName, String indexType, String eventUUID, String routingValue) throws ExecutionException {
     	initBulkRequestBuilder();
-        IndexRequestBuilder indexRequestBuilder = elasticSearchClientService.prepareIndex(indexName, indexType, eventUUID);
-        indexRequestBuilder.setSource(inputMessage, XContentType.JSON);
-        if (routingValue != null && routingValue.trim().length()>0) {
-            indexRequestBuilder.setRouting(routingValue);
+
+    	indexRequest = elasticSearchClientService.prepareIndexRequest(inputMessage,indexName,indexType,eventUUID);
+
```

Comment:
so we don't need to set content type as JSON?

Reply:
I'm setting XContentType.JSON later - in elasticSearchClientService.prepareIndexRequest:
 IndexRequest indexRequest = new IndexRequest().id(eventUUID).type(indexType).source(inputMessage, XContentType.JSON).index(indexName);